src/wp-includes/option.php

```diff
@@ -1106,6 +1113,71 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ * Determine the appropriate autoload value based on input.
+ *
+ * This function checks the provided autoload value and returns a standardized
+ * value ('yes', 'no', 'default-yes', or 'default-no') based on specific conditions.
+ *
+ * @since 6.4.0
+ *
+ * @param mixed $autoload The autoload value to evaluate.
+ *
+ * @return string The standardized autoload value: 'yes', 'no', 'default-yes', or 'default-no'.
+ */
+function get_autoload_value( $autoload ) {
+	// Check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'yes' : 'no';
+	}
+
+	switch ( $autoload ) {
+		case 'no':
+		case 'yes':
+		case 'default-yes':
+		case 'default-no':
+			return $autoload;
+		default:
+			return 'default-yes';
+	}
+}
```

Comment:
See my feedback above. This function is not needed, as the two `default-*` values should never be passed by developers and thus don't need "sanitization". If you want to avoid the duplicate logic of `$autoload = ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';`, then I think we should instead introduce a private helper like:
```php
function _sanitize_autoload_value( $autoload ) {
	if ( null === $autoload ) { // `null` is the default value which is resolved to a more concrete default separately.
		return null;
	}
	return ( 'no' === $autoload || false === $autoload ) ? 'no' : 'yes';
}
```

Also, I think we should reserve this kind of function name for something more prominent that could potentially help in the future. The function name sounds like it would return the autoload value of a specific option, but it's merely a very simple "sanitization" function.